tests/cycle_output.rs

```diff
@@ -136,7 +136,7 @@ fn revalidate_no_changes() {
 
     db.assert_logs(expect![[r#"
         [
-            "salsa_event(DidValidateMemoizedValue { database_key: read_value(Id(401)) })",
+            "salsa_event(DidValidateMemoizedValue { database_key: read_value(Id(403)) })",
```

Comment:
The `id` change here is because Salsa reused the IDs between iterations, even though the `Output` had different values. Now, Salsa keeps all intermediate constructed `Output` structs around.